chain/cosmos/chain_node.go

```diff
@@ -362,6 +361,94 @@ func (tn *ChainNode) FindTxs(ctx context.Context, height uint64) ([]blockdb.Tx,
 	return txs, nil
 }
 
+// TxCommand is a helper to retrieve a full command for broadcasting a tx
+// with the chain node binary.
+func (tn *ChainNode) TxCommand(keyName string, command ...string) []string {
+	command = append([]string{"tx"}, command...)
+	return tn.NodeCommand(append(command,
+		"--from", keyName,
+		"--gas-prices", tn.Chain.Config().GasPrices,
+		"--gas-adjustment", fmt.Sprint(tn.Chain.Config().GasAdjustment),
+		"--keyring-backend", keyring.BackendTest,
+		"--output", "json",
+		"-y",
+	)...)
+}
+
+// ExecTx executes a transaction, waits for 2 blocks if successful, then returns the tx hash.
+func (tn *ChainNode) ExecTx(ctx context.Context, keyName string, command ...string) (string, error) {
+	tn.lock.Lock()
+	defer tn.lock.Unlock()
+
+	stdout, _, err := tn.Exec(ctx, tn.TxCommand(keyName, command...), nil)
+	if err != nil {
+		return "", err
+	}
+	output := CosmosTx{}
+	err = json.Unmarshal([]byte(stdout), &output)
+	if err != nil {
+		return "", err
+	}
+	if output.Code != 0 {
+		return output.TxHash, fmt.Errorf("transaction failed with code %d: %s", output.Code, output.RawLog)
+	}
+	if err := test.WaitForBlocks(ctx, 2, tn); err != nil {
+		return "", err
+	}
+	return output.TxHash, nil
+}
+
+// NodeCommand is a helper to retrieve a full command for a chain node binary.
+// when interactions with the RPC endpoint are necessary.
+// For example, if chain node binary is `gaiad`, and desired command is `gaiad keys show key1`,
+// pass ("keys", "show", "key1") for command to return the full command.
+// Will include additional flags for node URL, home directory, and chain ID.
+func (tn *ChainNode) NodeCommand(command ...string) []string {
+	command = tn.BinCommand(command...)
+	return append(command,
+		"--node", fmt.Sprintf("tcp://%s:26657", tn.HostName()),
+		"--chain-id", tn.Chain.Config().ChainID,
+	)
+}
+
+// BinCommand is a helper to retrieve a full command for a chain node binary.
+// For example, if chain node binary is `gaiad`, and desired command is `gaiad keys show key1`,
+// pass ("keys", "show", "key1") for command to return the full command.
```

Comment:
This example is the same as the NodeCommand. Is that on purpose?

Reply:
yes, `NodeCommand` returns the command, `ExecNode` builds the command and also executes it.